ui/src/login/login.tsx

```diff
@@ -1,20 +1,55 @@
 import {Page} from 'argo-ui/src/components/page/page';
 import * as React from 'react';
+import {useState} from 'react';
 
 import {uiUrl, uiUrlWithParams} from '../shared/base';
+import {services} from '../shared/services';
 import {useCollectEvent} from '../shared/use-collect-event';
 
 import './login.scss';
 
+// Clear any cached data on logout
 function logout() {
     document.cookie = 'authorization=;Max-Age=0';
-    document.location.reload();
+    // Clear any cached API data
+    services.info.clearCache();
+    // Add a small delay before reload to ensure cookies are cleared
+    setTimeout(() => {
+        document.location.reload();
+    }, 100);
 }
+
+// Set auth token and redirect with loading indicator
 function user(token: string) {
+    if (!token || token.trim() === '') {
+        alert('Please enter a valid token');
+        return;
+    }
+
+    // Show loading indicator
+    const loadingElement = document.createElement('div');
+    loadingElement.className = 'loading-container';
+    loadingElement.innerHTML = `
+        <div style="text-align: center">
+            <div class="loading-spinner"></div>
+            <p>Logging in...</p>
+        </div>
+    `;
+    document.body.appendChild(loadingElement);
+
+    // Set the cookie
     const path = uiUrl('');
     document.cookie = 'authorization=' + token + ';SameSite=Strict;path=' + path;
-    document.location.href = path;
+
+    // Clear any cached API data
+    services.info.clearCache();
+
+    // Add a small delay before redirect to ensure cookie is set
```

Comment:
Could you explain why this is necessary? Setting a cookie is a synchronous operation.